src/Info.ts

```diff
@@ -0,0 +1,107 @@
+import { fileBackedObject } from "./FileBackedObject";
+import { SharedSettings } from "./SharedSettings";
+
+import Discord = require("discord.js");
+
+// If the Collection contains anything in the other array
+const findOne = (arr1: Discord.Collection<string, Discord.Role>, arr2: Array<any>) => {
+    return arr1.some(v => {
+        return arr2.indexOf(v) >= 0;
+    });
+};
+
+export interface InfoData {
+    Command: string;
+    Message: string;
+}
+
+export default class Info {
+    private bot: Discord.Client;
+    private infos: InfoData[];
+    private sharedSettings: SharedSettings;
+
+    constructor(bot: Discord.Client, sharedSettings: SharedSettings, userFile: string) {
+        console.log("Requested Thinking extension..");
+        this.bot = bot;
+
+        this.infos = fileBackedObject(userFile);
+        console.log("Successfully loaded info file.");
+
+        this.sharedSettings = sharedSettings;
+
+        this.bot.on("ready", this.onBot.bind(this));
+        this.bot.on("message", this.onInfo.bind(this));
+    }
+
+    onBot() {
+        console.log("Info extension loaded.");
+    }
+
+    onInfo(message: Discord.Message) {
+        if (!findOne(message.member.roles, this.sharedSettings.info.allowedRoles)) return;
+
+        const split: string[] = message.cleanContent.split(" ");
```

Comment:
You shouldn't need to have the `: string[]` type assertion here.

Reply:
👍 I'll remove it